pandas/core/frame.py

```diff
@@ -3533,12 +3533,23 @@ def _combine_match_columns(self, other, func, level=None, fill_value=None):
                                    axes=[left.columns, self.index])
         return self._constructor(new_data)
 
-    def _combine_const(self, other, func, raise_on_error=True):
+    def _combine_const(self, other, func,
+                       raise_on_error=True, fill_value=None):
         if self.empty:
             return self
 
-        new_data = self._data.eval(func=func, other=other,
-                                   raise_on_error=raise_on_error)
+        if fill_value is not None:
```

Comment:
the steps are ``align``, ``arith`` then ``fill`` (with mask).

``align`` is moot in this case. yes you are right, fill then arith.

```
In [7]: Series([np.nan,1,np.nan]).add(Series([2,3],index=[2,3]),fill_value=10)
Out[7]: 
0     NaN
1    11.0
2    12.0
3    13.0
dtype: float64
```